### DISCUSSION TITLE
Can't create interface Tagged VLANs via API

### DISCUSSION THREAD

**Discussion by Oussema-99** [netbox-community/netbox#13584] [Help Wanted!] [answered]
### NetBox version

v3.3.4

### Python version

3.8

### Steps to Reproduce

I have a device already created with all of its physical interfaces. I'm trying to update (Patch method) one of those interfaces to add the Tagged VLANs that it's supposed to have.
This is the Payload for my patch request:
`{'name': 'Eth1/1', 'device': {'name': 'DEVICE01'}, 'mtu': 9216, 'description': 'BogusDescription', 'mode': 'tagged', 'tagged_vlans': [{'id': 185, 'vid': 100, 'name': 'BogusName'}, {'id': 186, 'vid': 200, 'name': 'BogusName2'}]}`

### Expected Behavior

Tagged VLANs (and the other items) are added to the interface

### Observed Behavior

Request fails with the following Exception: 
https://192.168.x.x/api/dcim/interfaces/5465/ Returned the following Response: 400 - {
    "tagged_vlans": [
        "Incorrect type. Expected pk value, received dict."
    ]

**Comment by candlerb** [comment, marked as answer]
As the error says, the tagged_vlans attribute should be a list of primary key values (integers), not dicts.

```
"tagged_vlans": [185, 186]
```

**Comment by Oussema-99** [comment]
Thank you for clarifying. Looking at the API docs in the responses section it displays the entire VLAN Object in a dictionary hence the confusion on my part (especially since it works for all other items).

![image](https://github.com/netbox-community/netbox/assets/66390427/a749d598-c482-4515-8e3b-c0d6e93bf2ce)

**Reply by candlerb** [reply]
What matters is the request body, not the response body.

Here is what Netbox v3.5.9 shows for the PATCH request you're using:

![image](https://github.com/netbox-community/netbox/assets/44789/4b9d47de-dc72-4048-9899-aa3da03349e1)

However, I believe the REST API did get an overhaul in v3.5, so this is likely different to what you see with v3.3.

**Comment by Oussema-99** [comment]
Understandable, thank you very much for your timely assistance!
